ttnn/cpp/ttnn/operations/experimental/reshape/reshape.hpp

```diff
@@ -0,0 +1,27 @@
+// SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include "ttnn/run_operation.hpp"
+#include "ttnn/decorators.hpp"
+#include <optional>
+
+namespace ttnn {
+namespace operations::experimental::reshape {
+
+
+struct ReshapeOperation {
+    static ttnn::Tensor invoke(const ttnn::Tensor& input_tensor, const ttnn::Shape& shape);
+    static ttnn::Tensor invoke(const ttnn::Tensor& input_tensor, const ttnn::SimpleShape& shape);
+};
+
+}  // namespace operations::experimental::reshape
+
+namespace experimental {
+constexpr auto unsafe_view = ttnn::register_operation_with_auto_launch_op<
+    "ttnn::experimental::unsafe_view",
+    ttnn::operations::experimental::reshape::ReshapeOperation>();
```

Comment:
You call the struct ReshapeOperation, but the operation is `unsafe_view`. 
Please align.

Also, can you share a bit on why is this an `unsafe_view`? Is it because you guarantee "0 data movement or throw"?

Reply:
the unsafe_view is used when you only need a change in the metadata so 0 data movement. The name has changed from Tensor.reshape to unsafe_view to avoid the confusion with ttnn.reshape that does the actual reshape.